bundles/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/handler/MyQAccountHandler.java

```diff
@@ -267,10 +269,11 @@ private void getDevices() throws InterruptedException {
                 devices.items.forEach(device -> {
                     ThingTypeUID thingTypeUID = new ThingTypeUID(BINDING_ID, device.deviceFamily);
                     if (SUPPORTED_DISCOVERY_THING_TYPES_UIDS.contains(thingTypeUID)) {
-                        String id = device.serialNumber.toLowerCase();
+                        String deviceSerial = device.serialNumber.toLowerCase();
                         for (Thing thing : getThing().getThings()) {
                             // support both upper and lower case serialIds
-                            if (thing.getUID().getId().toLowerCase().equals(id)) {
+                            if (thing.getConfiguration().as(MyQDeviceConfiguration.class).serialNumber.toLowerCase()
+                                    .equals(deviceSerial)) {
```

Comment:
In general it's not recommend to get the configuration everytime. I would suggest to make a method `getSerialNumber()` on the handler and also use `equalsIgnoreCase`. Something like:
```suggestion
                            if (((MyQDeviceHandler) thing.getHandler()).getSerialNumber().equalsIgnoreCase(deviceSerial)) {
```

Reply:
>In general it's not recommend to get the configuration everytime. 

Yep agreed

>on the handler and also use equalsIgnoreCase

yeah, i'm not sure what i was thinking there

Reply:
done!